packages/Core/iterable.candy

```diff
@@ -157,6 +176,20 @@ forEach iterable action :=
 
 ## Conversion
 
+fromList l :=
+  needs (list.is l)
+  # () => Result[(Item, Iterable[Item]), Any]
+  # iterable | function.run | result.map { it -> it | list.get 0 }
+  # The state contains the list and an index.
+  newWithState (l, 0) { state ->
+    [l, index] = state
```

Comment:
This looks mismatched – the initial state is a list but we destructure a struct. Intuitively, I'd use a struct here.